### PR TITLE
Added code for reading SQS for LED Sign

### PR DESCRIPTION
**Pull request by andrewtedi**
reading rainbow

### CODE REVIEW

**Comment by allen-tran** [comment]
you are so awesome at coding

File: ledsign/sqsread.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+const express = require('express');
+const axios = require('axios');
+const app = express();
+const http = require("http");
+const hostname = 'X.X.X.X';
```
Comment (evanugarte) [inline ledsign/sqsread.js]: put host name and LED_URL in config.js

File: ledsign/sqsread.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    .then(response => {
+      res.send(response.data);
+    })
+    .catch(error => {
+      console.error('ERROR', error);
+    });
+});
+
+//listen for request on port 8083, and as a callback function have the port listened on logged
+server.listen(port, hostname, () => {
```
Comment (evanugarte) [inline ledsign/sqsread.js]: I think you can remove all the server stuff. this file just listens for messages pushed on SQS and then acts accordingly.

File: ledsign/sqsread.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  MaxNumberOfMessages: 1,
+  VisibilityTimeout: 0,
+  WaitTimeSeconds: 0
+};
+setInterval(() => {
+
+  sqs.receiveMessage(params, (err, data) => {
+
+    if (err) {
+      console.log(err, err.stack);
```
Comment (evanugarte) [inline ledsign/sqsread.js]:
```suggestion
      return;
```

File: ledsign/sqsread.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    } else {
+      // console.log(data, "whaa");
+      if (!data.Messages) {
+        console.log('Nothing to process');
```
Comment (evanugarte) [inline ledsign/sqsread.js]:
```suggestion
```

File: ledsign/sqsread.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        return;
+      }
+      // console.log("I am going to parse", data.Messages[0].Body);
+      const orderData = JSON.parse(data.Messages[0].Body);
+      console.log('Order received', orderData);
+      axios.post(LED_URL + 'api/update-sign', orderData);
+      // orderData is now an object that contains order_id and date properties
```
Comment (evanugarte) [inline ledsign/sqsread.js]: remove these comments

File: ledsign/sqsread.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      sqs.deleteMessage(deleteParams, (err, data) => {
+        if (err) {
+          console.log(err, err.stack);
+        } else {
+          console.log('Successfully deleted message from queue');
+        }
+      });
+    }
+  });
+}, 1000);
```
Comment (evanugarte) [inline ledsign/sqsread.js]: We will getting billed for querying the queue too much. let's do every 10 seconds instead
```suggestion
}, 10000);

```

File: ledsign/sqsread.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      // orderData is now an object that contains order_id and date properties
+      // Lookup order data from data storage
+      // Execute billing for order
+      // Update data storage
+      // Now we must delete the message so we don't handle it again
+      const deleteParams = {
+        QueueUrl: queueUrl,
+        ReceiptHandle: data.Messages[0].ReceiptHandle
+      };
+      sqs.deleteMessage(deleteParams, (err, data) => {
```
Comment (evanugarte) [inline ledsign/sqsread.js]: you can remove the if/else below and just make it an empty function, later we can have smarter error handling

File: ledsign/sqsread.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+    if (err) {
+      console.log(err, err.stack);
+    } else {
+      // console.log(data, "whaa");
+      if (!data.Messages) {
+        console.log('Nothing to process');
+        return;
+      }
+      // console.log("I am going to parse", data.Messages[0].Body);
```
Comment (evanugarte) [inline ledsign/sqsread.js]:
```suggestion
```

File: ledsign/sqsread.js [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+      const orderData = JSON.parse(data.Messages[0].Body);
+      console.log('Order received', orderData);
```
Comment (evanugarte) [inline ledsign/sqsread.js]: remove all console 🪵 
```suggestion
```

**Review by evanugarte — CHANGES_REQUESTED** [review]

**Review by evanugarte — APPROVED** [review]

**Review by allen-tran — APPROVED** [review]

**Pull request merged**